Update requirements.txt

Julian-Hochhaus opened the pull request:
As mentioned in #3 I have added all missing packages which were common on WSL install and Windows install to the `requirements.txt`

I do not see so far any version sensitive code happening for the 5 packages but feel free to edit the request and set defined package versions.

Julian-Hochhaus commented:
In addition, I have modified the standard values of the number of iterations.
In my experience, convergence with most models and optimization algorithms is reached after ~100-150 iterations max in complex cases. Most of the time, ~50 iterations are enough. 
I therefore set the minimum of the slider to 1 to not constrain the user too much and the initial value to 3, with the number of `max_nfev`, which is initially set to 50 by the `max_iterations` variabl,e this would translate to 150 iteration,s which I consider a reasonable value for complex models

Julian-Hochhaus commented:
I have fixed a bug that some variables of the config were never initialized if there was no config file was yet created.

gkerherve commented:
Hi Julian,
This does not work well if you have more than 1 doublet. I have just tried with Ceria (10 peaks) and I had to press the button 10 times to get a good. I think it is not an issue for people to wait until the 20 times. It may also be maybe with the constraint that are read after each convergence (here every 50 times). I would be happy to consider a value that could be set in the preference window but for now I would leave to how it was set as it works for most complicated fit

Maybe I have missed something else

From: Julian-Hochhaus ***@***.***>
Sent: 20 June 2025 10:21
To: KherveFitting/KherveFitting ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [KherveFitting/KherveFitting] Update requirements.txt (PR #6)

This email from ***@***.******@***.***> originates from outside Imperial. Do not click on links and attachments unless you recognise the sender. If you trust the sender, add them to your safe senders list<https://spam.ic.ac.uk/SpamConsole/Senders.aspx> to disable email stamping for this address.


[https://avatars.githubusercontent.com/u/22680429?s=20&v=4]Julian-Hochhaus left a comment (KherveFitting/KherveFitting#6)<https://github.com/KherveFitting/KherveFitting/pull/6#issuecomment-2990437202>

In addition, I have modified the standard values of the number of iterations.
In my experience, convergence with most models and optimization algorithms is reached after ~100-150 iterations max in complex cases. Most of the time, ~50 iterations are enough.
I therefore set the minimum of the slider to 1 to not constrain the user too much and the initial value to 3, with the number of max_nfev, which is initially set to 50 by the max_iterations variabl,e this would translate to 150 iteration,s which I consider a reasonable value for complex models

—
Reply to this email directly, view it on GitHub<https://github.com/KherveFitting/KherveFitting/pull/6#issuecomment-2990437202>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AKBYS2HTPP37YRBCTAPBRC33EPG7LAVCNFSM6AAAAAB7XVTL42VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSOJQGQZTOMRQGI>.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.******@***.***>>

Julian-Hochhaus commented:
Ok, I will update the  pull request to only include the bug fixing and remove the modified iterations

Julian-Hochhaus commented:
I hope the pull request now better matches your expectations. I have reduced it back to only include the bug fixes:

1. Creating config.json if it was never created before, and setting some standard values that were missing before
2. Added several requirements, which were missing on Linux as well as on Windows, to the 'requirements.txt'
3.  Modified the '.gitignore' so that temporary folders, such as the Backup folder of the user, are not included in the git pushes.

Julian-Hochhaus commented:
Hi Gwilherm,

No worries at all! Totally understand wanting to go through it with your brother first. As a physicist who spends more time searching UHV leaks than writing code, I’m also cautious with Git — always slightly worried I might accidentally delete something. :D

Really appreciate that you’ve already included the other requests!

Julian-Hochhaus commented:
Closing, as in the meantime, everything was merged through other pull requests.
